include/seqan3/core/type_traits/basic.hpp

```diff
@@ -137,6 +138,33 @@ using ignore_t = remove_cvref_t<decltype(std::ignore)>;
 template <typename t>
 constexpr bool decays_to_ignore_v = std::is_same_v<remove_cvref_t<t>, ignore_t>;
 
+// ----------------------------------------------------------------------------
+// strip_type_identity
+// ----------------------------------------------------------------------------
+
+/*!\brief A transformation trait that wraps a type in std::type_identity unless it is already a std::type_identity.
+ * \implements seqan3::transformation_trait
+ * \tparam t The type to operate on.
+ */
+template <typename t>
+struct strip_type_identity : std::type_identity<t>
```

Comment:
This one is tricky. I am actually only ever interested in `strip_type_identiy_t` (the shortcut). And the shortcut does exactly what it says, it **removes** type_identity if the type is wrapped and returns the type itself otherwise.
To implement said behaviour what I need to do is actually **add** type_identity to types (but only those that that aren't wrapped already).

I really don't know how to improve this :man_shrugging:

Reply:
But wow, this is confusing. Ok' lets say you are only interested in strip_type_identity, why should this not work?
```cpp
template <typename t>
struct strip_type_identity
{
    using type = std::conditional_t<detail::is_type_specialisation_of_v<t, std::type_identity>, 
                                    typename t::type,
                                    t>;
};
```

Reply:
Your example results in exactly the same semantics as my code :rofl: 

Think about it: Your `strip_type_identity` exposes a member `type` that is `t` -> this is the type_identity! So it wraps everything in type_identity except those types that are already wrapped, they are unwrapped and wrapped again :wink: 

What would work would be using the conditional directly in the shortcut and not having any class at all. However `typename t::type` is not defined for all types, so I would need to use another metafunction or some lazy mechanism...
I will investigate it, but I am not sure other solutions will make this more readable.

Reply:
>Your example results in exactly the same semantics as my code 

Of course it does. That's the whole point. 

> Think about it: Your strip_type_identity exposes a member type that is t -> this is the type_identity! So it wraps everything in type_identity except those types that are already wrapped, they are unwrapped and wrapped again 

Not sure, what you are implying but now the function does what the name suggests instead of the previous version where it did the exact opposite. Namely, wrapping the type in  type_identity if it wasn't already. I mean the whole point of this MF ist to have a generic solution for type_identity and non type_identity types, isn't it?

Anyway, I wouldn't mind just having the shortcut. I also think that this is better placed in a detail namespace for now. Or are you exposing it somewhere public? 
With `typename t::type` you could use `transformation_trait_or_t<t, void>` I guess. That would be instantiable for any type but would only be picked if in fact it is a transformation trait. Maybe there is even a more elegant solution.

Reply:
It is in namespace detail. I remove the trait so thereis only a shortcut.